libs/ledger-live-common/src/currencies/sortByMarketcap.ts

```diff
@@ -31,23 +31,32 @@ function lazyLoadTickerMap() {
   return m;
 }
 
+function getCurrencyId(currency: Currency): string {
+  if ("id" in currency && currency.id) {
+    return currency.id;
+  }
+  return currency.ticker;
+}
+
 export const sortByMarketcap = <C extends Currency>(currencies: C[], tickers: string[]): C[] => {
-  const m = lazyLoadTickerMap();
-  const list = currencies.slice(0);
-  const prependList: C[] = [];
-  tickers.forEach(ticker => {
-    const item: C | undefined = m.get(ticker) as C | undefined;
+  const tickerMap = lazyLoadTickerMap();
+  const sortedCurrenciesMap: Map<string, C> = new Map();
+  const remainingCurrencies: C[] = [];
 
-    if (item) {
-      const i = list.indexOf(item);
+  for (const ticker of tickers) {
+    const currency = tickerMap.get(ticker) as C | undefined;
+    if (currency) {
+      sortedCurrenciesMap.set(getCurrencyId(currency), currency);
+    }
+  }
 
-      if (i !== -1) {
-        list.splice(i, 1);
-        prependList.push(item);
-      }
+  currencies.forEach(currency => {
```

Comment:
could you also use `for of` syntax here so we have similar code style as the previous for loop?

alternatively it seems the logic here is also fundamentally a
```
const remainingCurrencies = currencies.filter(c => !sortedCurrenciesMap.has(getCurrencyId(c)));
```
which may be similar in perf but simpler code ? 🤔